src/apple/alohalytics_objc.mm

```diff
@@ -266,6 +265,58 @@ static void LogSystemInformation() {
   }
   return parsed;
 }
+
+// Need it to effectively upload data when app goes into background.
+static UIBackgroundTaskIdentifier sBackgroundTaskId = UIBackgroundTaskInvalid;
+static void EndBackgroundTask() {
+  [[UIApplication sharedApplication] endBackgroundTask:sBackgroundTaskId];
+  sBackgroundTaskId = UIBackgroundTaskInvalid;
+}
+static void OnUploadFinished(alohalytics::ProcessingResult result) {
+  if (Stats::Instance().DebugMode()) {
+    const char * str;
+    switch (result) {
+      case alohalytics::ProcessingResult::ENothingToProcess: str = "There is no data to upload."; break;
+      case alohalytics::ProcessingResult::EProcessedSuccessfully: str = "Data was uploaded successfully."; break;
+      case alohalytics::ProcessingResult::EProcessingError: str = "Error while uploading data."; break;
+    }
```

Comment:
It is missing intentionally - if new enum value will be added, we get a compiler warning.

> On Jun 10, 2015, at 19:51, Dima <notifications@github.com> wrote:
> 
> In src/apple/alohalytics_objc.mm <https://github.com/KnowSheet/Alohalytics/pull/36#discussion_r32139070>:
> 
> > @@ -266,6 +265,58 @@ static void LogSystemInformation() {
> >    }
> >    return parsed;
> >  }
> > +
> > +// Need it to effectively upload data when app goes into background.
> > +static UIBackgroundTaskIdentifier sBackgroundTaskId = UIBackgroundTaskInvalid;
> > +static void EndBackgroundTask() {
> > +  [[UIApplication sharedApplication] endBackgroundTask:sBackgroundTaskId];
> > +  sBackgroundTaskId = UIBackgroundTaskInvalid;
> > +}
> > +static void OnUploadFinished(alohalytics::ProcessingResult result) {
> > +  if (Stats::Instance().DebugMode()) {
> > +    const char * str;
> > +    switch (result) {
> > +      case alohalytics::ProcessingResult::ENothingToProcess: str = "There is no data to upload."; break;
> > +      case alohalytics::ProcessingResult::EProcessedSuccessfully: str = "Data was uploaded successfully."; break;
> > +      case alohalytics::ProcessingResult::EProcessingError: str = "Error while uploading data."; break;
> > +    }
> Missing default:.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/KnowSheet/Alohalytics/pull/36/files#r32139070>.
>